src/yaml2candid/src/lib.rs

```diff
@@ -67,8 +84,8 @@ impl Yaml2Candid {
     /// * YAML types do not match the IDL types.
     /// * `type_name` is not defined in the converter's `.did` file.
     pub fn convert(&self, typ: &IDLType, data: &YamlValue) -> anyhow::Result<IDLValue> {
```

Comment:
Here the match includes just the Candid type, so that the compiler will complain if there are any unsupported Candid types.

Below _many_ types that were previously unsupported have been added.